scripts/background.js

```diff
@@ -152,6 +162,110 @@ function FetchSteamUserData( callback )
 	} );
 }
 
+function FetchSteamUserFamilyData( callback )
+{
+	if( userFamilyDataCache !== null )
+	{
+		callback( { data: userFamilyDataCache } );
+		return;
+	}
+
+	fetch( `https://store.steampowered.com/pointssummary/ajaxgetasyncconfig`, {
+		credentials: 'include',
+		headers: {
+			Accept: 'application/json',
+		},
+	} ).then( ( response ) => response.json() )
+		.then( ( response ) =>
+		{
+			if( !response || !response.success || !response.data || !response.data.webapi_token )
+			{
+				throw new Error( 'Are you logged on the Steam Store in this browser?' );
+			}
+
+			const accessToken = response.data.webapi_token;
+
+			if( accessToken )
+			{
+				const paramsGroupId = new URLSearchParams();
+				paramsGroupId.set( 'access_token', accessToken );
+				fetch( `https://api.steampowered.com/IFamilyGroupsService/GetFamilyGroupForUser/v1/?${paramsGroupId.toString()}`, {
+					headers: {
+						Accept: 'application/json',
+					},
+				} ).then( ( response ) =>  response.json() )
+					.then( ( response ) =>
+					{
+						if( !response || !response.response )
+						{
+							throw new Error( 'Is Steam okay?' );
+						}
+						else if( response.response.is_not_member_of_any_group || !response.response.family_groupid )
+						{
+							throw new Error( 'You are not part of a family group.' );
+						}
+
+						return response.response.family_groupid ;
+					} )
+					.then( ( family_groupid ) =>
+					{
+						const paramsSharedLibrary = new URLSearchParams();
+						paramsSharedLibrary.set( 'access_token', accessToken );
+						paramsSharedLibrary.set( 'include_free', 'true' );
+						paramsSharedLibrary.set( 'family_groupid', family_groupid );
+						paramsSharedLibrary.set( 'include_own', 'true' );
+						paramsSharedLibrary.set( 'include_non_games', 'true' );
+						// the include_own param has no link with its name, if set at false, it returns only your owned apps, if set at true, it returns your owned apps and the apps from your family
+						fetch( `https://api.steampowered.com/IFamilyGroupsService/GetSharedLibraryApps/v1/?${paramsSharedLibrary.toString()}`, {
+							headers: {
+								Accept: 'application/json',
+							}
+						} ).then( ( response ) => response.json() )
+							.then( ( response ) =>
+							{
+								if( !response || !response.response || !response.response.apps )
+								{
+									throw new Error( 'Is Steam okay?' );
+								}
+								const reduced = response.response.apps.reduce( ( data, app ) =>
+								{
+									if( !app.owner_steamids.includes( response.response.owner_steamid ) )
+									{
+										data[ app.appid ] = app.owner_steamids;
```

Comment:
Not sure why steamid needs to be stored, just a plain list of appids should be fine.